### PR TITLE
add release note callback functionality

### PR DESCRIPTION
**Pull request by RaphaelVogel**

### CODE REVIEW

File: concourse/model/traits/release.py [hunk 1]
```diff
@@ -128,6 +128,16 @@ def _attribute_specs(cls):
         Any changes left inside the worktree are added to the resulting release commit.
         ''',
     ),
+    AttributeSpec.optional(
+        name='release_note_callback',
```
Comment (ccwienk) [inline concourse/model/traits/release.py]: nit: release_notes_callback

File: concourse/model/traits/release.py [hunk 2]
```diff
@@ -164,6 +174,9 @@ def release_callback_path(self):
     def next_version_callback_path(self):
         return self.raw['next_version_callback']
 
+    def release_note_callback_path(self):
+        return self.raw['release_note_callback']
```
Comment (ccwienk) [inline concourse/model/traits/release.py]: ditto

File: concourse/model/traits/release.py [hunk 3]
```diff
@@ -128,6 +128,16 @@ def _attribute_specs(cls):
         Any changes left inside the worktree are added to the resulting release commit.
         ''',
     ),
+    AttributeSpec.optional(
+        name='release_note_callback',
+        default=None,
+        doc='''
+        an optional callback that is called during release notes creation.
```
Comment (ccwienk) [inline concourse/model/traits/release.py]: document:
- path is relative to repo root
- file must be executable

File: concourse/model/traits/release.py [hunk 4]
```diff
@@ -128,6 +128,16 @@ def _attribute_specs(cls):
...
+        The callback is passed the absolute path to the releases notes via environment
```
Comment (ccwienk) [inline concourse/model/traits/release.py]: state that the release notes are a regular (text) file with utf-8 encoding?

File: concourse/steps/release.mako [hunk 5]
```diff
@@ -67,6 +68,9 @@ release_and_prepare_next_dev_cycle(
   % if next_version_callback_path:
   next_version_callback='${next_version_callback_path}',
   % endif
+  % if release_note_callback_path:
+  release_note_callback='${release_note_callback_path}'
```
Comment (ccwienk) [inline concourse/steps/release.mako]: notes?

**Comment by RaphaelVogel** [comment]
Not needed by test machinery anymore.

**Pull request closed without merging**